internal/server/session.go

```diff
@@ -71,8 +71,12 @@ func (session *Session) PrepareServerCxxCmdLine(noccServer *NoccServer, clientCw
 		serverIdir := session.client.MapClientFileNameToServerAbs(cxxIDirs[i+1])
 		cxxCmdLine = append(cxxCmdLine, arg, serverIdir)
 	}
-	// append -Wall and other cxx args
-	cxxCmdLine = append(cxxCmdLine, cxxArgs...)
+
+	for i := 0; i < len(cxxArgs); i++ {
+		cxxArg := FilePrefixMapOption(cxxArgs[i], session.client.workingDir)
+
+		cxxCmdLine = append(cxxCmdLine, cxxArg)
+	}
```

Comment:
Does not handle the case where the client does not specify `-ffile-prefix-map`

P.S. Search seems to have quadratic complexity, but since cmdline is not explected to be long, it might be not an issue...